src/test/util_tests.cpp

```diff
@@ -1791,4 +1791,13 @@ BOOST_AUTO_TEST_CASE(util_WriteBinaryFile)
     BOOST_CHECK(valid);
     BOOST_CHECK_EQUAL(actual_text, expected_text);
 }
+
+BOOST_AUTO_TEST_CASE(clearshrink_test)
+{
+    std::vector<uint8_t> v = {1, 2, 3};
```

Comment:
Given its specialization, would adding a test on `std::vector<bool>` be worthwhile?